addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshEvent.java

```diff
@@ -0,0 +1,171 @@
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.db.DatabaseException;
+import org.parosproxy.paros.network.HttpBody;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.zaproxy.addon.oast.OastRequest;
+import org.zaproxy.zap.network.HttpRequestBody;
+import org.zaproxy.zap.network.HttpResponseBody;
+
+public class InteractshEvent {
+
+    public static final Logger LOGGER = LogManager.getLogger(InteractshEvent.class);
+    static final String EMAIL_FROM_HEADER = "Email-From";
+
+    private final String protocol;
+    private final String uniqueId;
+    private final String fullId;
+    private final String rawRequest;
+    private final String rawResponse;
+    private final String remoteAddress;
+    private final Instant timestamp;
+    private final String queryType;
+    private final String smtpFrom;
+
+    public InteractshEvent(
+            String protocol,
+            String uniqueId,
+            String fullId,
+            String rawRequest,
+            String rawResponse,
+            String remoteAddress,
+            String timestamp,
+            String queryType,
+            String smtpFrom) {
+        this.protocol = protocol;
+        this.uniqueId = uniqueId;
+        this.fullId = fullId;
+        this.rawRequest = rawRequest;
+        this.rawResponse = rawResponse;
+        this.remoteAddress = remoteAddress;
+        this.timestamp = Instant.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(timestamp));
+        this.queryType = queryType;
+        this.smtpFrom = smtpFrom;
+    }
+
+    public InteractshEvent(JSONObject event) {
+        this(
+                event.getString("protocol"),
+                event.getString("unique-id"),
+                event.getString("full-id"),
+                event.getString("raw-request"),
+                event.optString("raw-response"),
+                event.optString("remote-address"),
+                event.getString("timestamp"),
+                event.optString("q-type"),
+                event.optString("smtp-from"));
+    }
+
+    public OastRequest toOastRequest() throws HttpMalformedHeaderException, DatabaseException {
+        HttpMessage msg =
+                new HttpMessage(
+                        new HttpRequestHeader(),
+                        new HttpRequestBody(),
+                        new HttpResponseHeader(),
+                        new HttpResponseBody());
+        msg.setTimeSentMillis(timestamp.toEpochMilli());
+        if ("http".equals(protocol)) {
+            extractAndSetMsg(msg.getRequestHeader(), msg.getRequestBody(), rawRequest);
+            extractAndSetMsg(msg.getResponseHeader(), msg.getResponseBody(), rawResponse);
+        } else if ("dns".equals(protocol)) {
```

Comment:
Do these conditions need to ignore case?

Reply:
More below too.

Reply:
Nope, this is how they are always sent, e.g. see [this link](https://github.com/projectdiscovery/interactsh/blob/f5b53a7b8be329ee6d4936ed753e3c46472e5726/pkg/server/http_server.go#L100) for "http".